Add WSL2 installation docs, fixes #2320 (#2321)
drud/ddev: docs/index.md

* Install a WSL2 distro, e.g. [Ubuntu 20.04](https://www.microsoft.com/store/productId/9N6SVWS3RX71). Install Ubuntu (or any other distro) _before_ installing Docker, so that Ubuntu becomes the default WSL distro.
* Install Docker Desktop for Windows (versions 2.3.0.2 and higher automatically use the WSL2 backend): [download Docker Desktop for Windows from Docker Hub](https://hub.docker.com/editions/community/docker-ce-desktop-windows/).
* Go to Docker Desktop settings > Resources > WSL integration > enable integration for your distro (now `docker` commands will be available from within your WSL2 distro).
* Install [Chocolatey](https://chocolatey.org/install) on Windows.
* Open a git-bash terminal with administrator rights and install mkcert by running `choco install mkcert`
* In git-bash, run `mkcert -install` and answer the prompt allowing the installation of the Certificate Authority.
* In git-bash, run the command `setx CAROOT "$(mkcert -CAROOT)" && setx WSLENV "CAROOT/p;$WSLENV"`. This will set WSL2 to use the Certificate Authority installed on the Windows side.
* Open the Ubuntu 20.04 terminal from the Windows start menu.

> * Run `export CAROOT=/mnt/c/Users/YOUR_WINDOWS_USERNAME/AppData/Local/mkcert`. This will make sure that `mkcert` on Linux uses your Windows certificates when issuing SSL certificates, so you don't get SSL warnings in your browser on Windows.

Reviewer: If you set the variable on the Windows side and expose it to WSL like done above this is not needed anymore

Reviewer 2: Thought I had already deleted that line, thanks.

Take the steps below to install DDEV into WSL2.

* Install WSL2 by [following Microsoft's docs](https://docs.microsoft.com/en-us/windows/wsl/install-win10). Make sure the WSL version is set to 2: `wsl --set-default-version 2`
* Install a WSL2 distro, e.g. [Ubuntu 20.04](https://www.microsoft.com/store/productId/9N6SVWS3RX71). Install Ubuntu (or any other distro) _before_ installing Docker, so that Ubuntu becomes the default WSL distro.
* Install Docker Desktop for Windows (versions 2.3.0.2 and higher automatically use the WSL2 backend): [download Docker Desktop for Windows from Docker Hub](https://hub.docker.com/editions/community/docker-ce-desktop-windows/).
* Go to Docker Desktop settings > Resources > WSL integration > enable integration for your distro (now `docker` commands will be available from within your WSL2 distro).
* Install [Chocolatey](https://chocolatey.org/install) on Windows.

> * Open a PowerShell terminal with administrator rights and run `choco install mkcert`

Reviewer 2: ```suggestion
* Open a git-bash or PowerShell terminal with administrator rights and install mkcert by running `choco install mkcert`
```

* Run `mkcert -install`. Your certificates are installed in `%localappdata%\mkcert`, we'll need these later on.
* Open the Ubuntu 20.04 terminal from the Windows start menu.
* Run `export CAROOT=/mnt/c/Users/YOUR_WINDOWS_USERNAME/AppData/Local/mkcert`. This will make sure that `mkcert` on Linux uses your Windows certificates when issuing SSL certificates, so you don't get SSL warnings in your browser on Windows.
* Run `echo 'export CAROOT=/mnt/c/Users/YOUR_WINDOWS_USERNAME/AppData/Local/mkcert' >> ~/.profile` to store this in your profile for future sessions.
* Now we'll install DDEV **for Linux** within WSL2. Open your WSL2 distro and follow the instructions for Linux below, **then come back here**.

> * After installation, run `mkcert -install` and you'll notice that mkcert will use your Windows CA certificates 🚀:

Reviewer 2: I don't think that any `mkcert -install` is required on the WSL2 side unless you're going to use X11 or a local tool that checks certs (like curl). But I guess it's totally reasonable to get it installed now.

  > Using the local CA at "/mnt/c/Users/YOUR_WINDOWS_USERNAME/AppData/Local/mkcert"
* That's it! You have now installed DDEV on WSL2 🎉 Remember to run all `ddev` commands in your Ubuntu/WSL2 terminal, **not** in PowerShell/Command Prompt.

> **Make sure you put your projects in the Linux filesystem (e.g. /home/LINUX_USERNAME), _not_ in the Windows filesystem (/mnt/c), for a much better performance.**

Reviewer 2: ```suggestion
**Make sure you put your projects in the Linux filesystem (e.g. /home/LINUX_USERNAME), _not_ in the Windows filesystem (/mnt/c), because you'll get vastly better performance on the Linux filesystem.**
```
